lib/application/pages/home/collections/update/update_collection_page.dart

```diff
@@ -0,0 +1,164 @@
+import 'package:memo/application/pages/home/collections/update/update_collection_memos.dart';
+import 'package:memo/application/pages/home/collections/update/update_providers.dart';
+import 'package:memo/application/theme/theme_controller.dart';
+import 'package:memo/application/view-models/home/update_collection_vm.dart';
+import 'package:memo/application/widgets/theme/custom_button.dart';
+import 'package:memo/application/widgets/theme/retry_container.dart';
+import 'package:memo/application/widgets/theme/themed_container.dart';
+import 'package:memo/application/widgets/theme/themed_tab_bar.dart';
+
+enum Segment { details, memos }
+
+class UpdateCollectionPage extends HookWidget {
+  @override
+  Widget build(BuildContext context) {
+    final vm = readUpdateCollectionVM(context);
+
+    final selectedSegment = useState(Segment.details);
+    final tabController = useTabController(initialLength: Segment.values.length);
+
+    useEffect(() {
+      void tabListener() => selectedSegment.value = Segment.values[tabController.index];
+
+      tabController.addListener(tabListener);
+      return () => tabController.removeListener(tabListener);
+    });
+
+    final tabs = Segment.values.map((segment) => Text(segment.title)).toList();
+    final title = vm.isEditing ? strings.editCollection : strings.newCollection;
+
+    return Scaffold(
+      appBar: AppBar(title: Text(title)),
+      body: Column(
+        children: [
+          ThemedTabBar(controller: tabController, tabs: tabs),
+          Expanded(child: _UpdateCollectionContents(selectedSegment: selectedSegment.value)),
+          const Spacer(),
+          _BottomActionContainer(
+            onSegmentSwapRequested: (segment) => tabController.animateTo(Segment.values.indexOf(segment)),
+            selectedSegment: selectedSegment.value,
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _UpdateCollectionContents extends HookWidget {
+  const _UpdateCollectionContents({required this.selectedSegment});
+
+  final Segment selectedSegment;
+
+  @override
+  Widget build(BuildContext context) {
+    final vm = readUpdateCollectionVM(context);
+    final state = useUpdateCollectionState();
+
+    if (state is UpdateCollectionFailedLoading) {
+      return Center(child: ExceptionRetryContainer(exception: state.exception, onRetry: vm.loadInitialContent));
+    }
+
+    switch (selectedSegment) {
+      case Segment.details:
+        return UpdateCollectionDetails();
+      case Segment.memos:
+        return UpdateCollectionMemos();
+    }
+  }
+}
+
+extension on Segment {
+  String get title {
+    switch (this) {
+      case Segment.details:
+        return strings.details;
+      case Segment.memos:
+        return strings.memos;
+    }
+  }
+}
+
+class _BottomActionContainer extends HookWidget {
+  const _BottomActionContainer({required this.selectedSegment, required this.onSegmentSwapRequested});
```

Comment:
Why nest all of this behavior in these widgets? It makes things much complex to understand. I would simply expose these functionalities to the "consumers" or the parent widgets, so that they can do whatever they want with actions made here in the container, and then, with that, change this bottom container to be simply configurable to all use-cases we need here

Reply:
I've split in different Widgets because the action button may assume different states based on the selected segment and the information that the user added. All use-cases are the following:

- When `Details` segment is selected the button is:
  - Enabled with `Create Collection` title when all the details information is complete and it has at least one memo;
  - Enabled with `Próximo` title when all details information is complete but the user doesn't have at least one memo;
  - Disabled with `Próximo` title when it doesn't have details information complete

- When `Memos` segment is selected the button is:
  - Enabled with `Create Collection` when `details` segment is complete and the user has at least one memo;
  - Disabled with `Create Collection` when there's missing information in `details` or the user doesn't have any memo created. 


Based on all these use-cases, I decided to create distinct buttons for each segment, since it's simpler to understand this division. If we move this use-cases scenarios to a single Widget, we would face an `if hell` scenario with a lot of ternary operations due to the different titles and actions.